### ISSUE TITLE
Invalid memory in frame.currentDocument.request

### ISSUE THREAD

**Issue by inancgumus** [grafana/xk6-browser#126]
After fixing #109 with #118, a new issue has popped up.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x1c3239f]
default   [--------------------------------------] 1 VUs  00m08.1s/10m0s  0/1 iters, 1 per VU
goroutine 85 [running]:
github.com/grafana/xk6-browser/common.(*Frame).clearLifecycle(0xc002b9c480)
        github.com/grafana/xk6-browser@v0.0.0-00010101000000-000000000000/common/frame.go:201 +0x17f
github.com/grafana/xk6-browser/common.(*FrameManager).frameNavigated(0xc001c9e120, {0xc00190efe0, 0x0}, {0x0, 0x0}, {0xc00190f000, 0x20}, {0x0, 0x0}, {0xc000d35f20, ...}, ...)
        github.com/grafana/xk6-browser@v0.0.0-00010101000000-000000000000/common/frame_manager.go:227 +0x588
github.com/grafana/xk6-browser/common.(*FrameSession).onFrameNavigated(0xc00138a480, 0xc000493f80, 0x20)
        github.com/grafana/xk6-browser@v0.0.0-00010101000000-000000000000/common/frame_session.go:509 +0x10d
github.com/grafana/xk6-browser/common.(*FrameSession).initEvents.func1()
        github.com/grafana/xk6-browser@v0.0.0-00010101000000-000000000000/common/frame_session.go:195 +0x3ec
created by github.com/grafana/xk6-browser/common.(*FrameSession).initEvents
        github.com/grafana/xk6-browser@v0.0.0-00010101000000-000000000000/common/frame_session.go:180 +0xe7
2021/11/18 17:17:40 [ERROR] exit status 2
```

This time, it's coming from here (`frame.currentDocument.request` becomes `nil`):

https://github.com/grafana/xk6-browser/blob/b7f55930199bebb70ee553bc53627f8965aca866/common/frame.go#L201

According to my research, `currentDocument`'s request can be nil. For example:

https://github.com/grafana/xk6-browser/blob/b7f55930199bebb70ee553bc53627f8965aca866/common/frame_manager.go#L197-L225

A fix is coming shortly...

**Comment by imiric** [comment]
This seems like a duplicate of #119, and I think I've seen the stack trace before that.

Why do you think it's related to #118?

**Comment by inancgumus** [comment]
> This seems like a duplicate of #119, and I think I've seen the stack trace before that.

119 is acting differently on my machine. It just hangs.

> Why do you think it's related to #118?

It started happening after #118 fix. So I feel like they are somehow related.

**Comment by imiric** [comment]
It hangs because of the `constant-arrival-rate` duration, you can remove `options` if you want. If it doesn't fail right away that means you're not able to reproduce it. You'll notice it's the same stack trace, +/- some line numbers which are different because of the recent changes. So I think it's a duplicate and we can close this issue.

> It started happening after #118 fix. So I feel like they are somehow related.

Does it happen more frequently since #118? What script are you using to reproduce it?

**Comment by inancgumus** [comment]
I removed the `options` and it works fine on my machine (on the `main` branch).

However, this issue is something different that I see every time I run Tom's oldest script in headless mode as I said in the #127's description.
